BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c

```diff
@@ -470,6 +469,13 @@ GatherDeviceInfo (
     PciIoDevice->Decodes = PciIoDevice->Parent->Decodes;
   }
 
+  EnumPolicy = (PCI_ENUM_POLICY_INFO *)PcdGetPtr (PcdPciEnumPolicyInfo);
+  if (!PciIoDevice->IsPciExp && (EnumPolicy->Downgrade.NonPcieDevices == 1)) {

```

Comment:
Actually, no specific use case. This is just to force Non-PCIe system devices to have downgraded BARs instead of re-assigning 32-bit BAR after PCI enumeration for 32-bit FSP. There will be still issue if 32-bit FSP accesses 64-bit PCIe BAR, but mostly not.

Reply:
OK, got it.